### ISSUE TITLE
Running any Windows container returns: docker: Error response from daemon: hcsshim::CreateComputeSystem ...: The request is not supported.

### ISSUE THREAD

**Issue by AntWatCDL** [microsoft/Windows-Containers#227]
**Describe the bug**
I am unable to run any windows containers on my PC.  A colleague gets the same behaviour.

**To Reproduce**
I get the following error:

“`docker: Error response from daemon: hcsshim::CreateComputeSystem ...: The request is not supported.`”

When I run the following test on my Windows 10 PC:

1) Install Docker Desktop 4.8.1.  Start Docker Desktop .
2) Switch to Windows Containers
3) In a PowerShell type > `docker image pull mcr.microsoft.com/windows/nanoserver:1809`
4) Then > `docker container run mcr.microsoft.com/windows/nanoserver:1809 hostname`

I am unable to run any windows containers on my PC.  A colleague gets the same behaviour.

**Expected behavior**
The Windows container runs.

**Configuration:**
 - Edition: Windows Version: 19044.1645
- Base Image being used: nanoserver
 - Container engine: Docker Desktop 4.8.1
 - Container Engine version : Docker engine v20.10.14

**Additional context**
I have confirmed Containers and HyperV are enabled.
Docker Desktop is using WSL2
Linux containers work ok.

**Comment by dcantah** [comment]
Curious, do any of the other nanoserver tags work? (1909, 2004 etc.)

**Comment by AntWatCDL** [comment]
I have tried all the following, with the same error each time:

mcr.microsoft.com/dotnet/framework/runtime:4.8-windowsservercore-ltsc2019

mcr.microsoft.com/windows/nanoserver:2004
mcr.microsoft.com/windows/nanoserver:20H2
mcr.microsoft.com/windows/nanoserver:1909
mcr.microsoft.com/windows/nanoserver:1809

**Comment by dcantah** [comment]
Okay, I'll try and setup a VM on your build and try this out today or tomorrow

**Comment by fady-azmy-msft** [comment]
Any updates on this you can share @dcantah?

**Comment by lippertmarkus** [comment]
The only time I had that error the issue was that the containers feature wasn't installed.

**Comment by dcantah** [comment]
@lippertmarkus The additional context portion states they made sure it was on at least, and sorry for the radio silence here.. Could you take a trace with this file and send it to dcanter@microsoft.com? https://github.com/MicrosoftDocs/Virtualization-Documentation/blob/main/windows-server-container-tools/wpr-profiles/HcsTraceProfile.wprp

So:

1. wpr -start hcstraceprofile.wprp
2. Repro 
3. wpr -stop trace.etl

**Comment by vzakusylo** [comment]
Hi, I've faced the same error, here is how to reproduce it via docker file 
--------------------------------------------------------------------------
FROM mcr.microsoft.com/windows/servercore/iis

ADD https://download.microsoft.com/download/C/6/D/C6D0FD4E-9E53-4897-9B91-836EBA2AACD3/vcredist_x86.exe /vcredist_x86_2010.exe 
RUN C:\vcredist_x86_2010.exe /quiet /install

ADD https://download.microsoft.com/download/2/E/6/2E61CFA4-993B-4DD4-91DA-3737CD5CD6E3/vcredist_x86.exe /vcredist_x86_2013.exe 
RUN C:\vcredist_x86_2013.exe /quiet /install

**Comment by vrapolinario** [comment]
@AntWatCDL can you please confirm you still see an issue here? By looking at your docker run command, I'm not sure what you mean by adding the "hostname". If this is related to environment variable, you need to add the -e parameter. Now, if this is regarding the network type, Windows containers don't support Host Network, so you'd have to use a different option. The default there is NAT.

@vzakusylo I have a few notes on the dockerfile you provided:
- The image you're using doesn't have a tag. Windows container images have a specific tag (not latest). You can see the list of available tags for IIS here: https://hub.docker.com/_/microsoft-windows-servercore-iis
- I'd recommend you use the RUN command to download the binary you need. That way you're calling a PowerShell session that can run Invoke-WebRequest. Your command could be: RUN PowerShell Invoke-WebRequest -Uri https://download.microsoft.com/download/C/6/D/C6D0FD4E-9E53-4897-9B91-836EBA2AACD3/vcredist_x86.exe -UseBasicParsing -OutFile vcredist_x86.exe

I have not tried this. In fact, can you please share what is the actual error message and at what point of the docker build it fails?

**Comment by Jeb135** [comment]
Are you running the powershell session on your host as an administrator? I noticed that I receive this error when I forget to open powershell as an administrator before trying to run the docker cli.

**Comment by sanme98** [comment]
I got the same error either run powershell as admin or not.

**Comment by fady-azmy-msft** [comment]
Closing this issue due to inactivity

**Comment by balindra-sahani-eptura** [comment]
Hi 
task: CosmosDbEmulator@2
  inputs:
    containerName: 'azure-cosmosdb-emulator'
    enableAPI: 'SQL'
    portMapping: '8081:8081, 8901:8901, 8902:8902, 8979:8979, 10250:10250, 10251:10251, 10252:10252, 10253:10253, 10254:10254, 10255:10255, 10256:10256, 10350:10350'
    hostDirectory: '$(Build.BinariesDirectory)\azure-cosmosdb-emulator'
    
**Note:** vImage : ubuntu-latest

    The above task gives the error
    
    Digest: sha256:ad6c3ba14cb41e1a71f61d97ea90d9309b0700242cd56c5e67958884448bc3b3
Status: Downloaded newer image for mcr.microsoft.com/cosmosdb/windows/azure-cosmos-emulator:latest
 container 9c3f1f436ec331838a8acc88bb1c32d2e041d099d42358c80e1b726dd59483bf created from image mcr.microsoft.com/cosmosdb/windows/azure-cosmos-emulator:latest
docker start 9c3f1f436ec331838a8acc88bb1c32d2e041d099d42358c80e1b726dd59483bf
**Error response from daemon: hcs::CreateComputeSystem 9c3f1f436ec331838a8acc88bb1c32d2e041d099d42358c80e1b726dd59483bf: The container operating system does not match the host operating system.**
Error: failed to start containers: 9c3f1f436ec331838a8acc88bb1c32d2e041d099d42358c80e1b726dd59483bf

Task failed: container 9c3f1f436ec331838a8acc88bb1c32d2e041d099d42358c80e1b726dd59483bf failed to start

**********************

Windows PowerShell output

**********************
Start-CosmosDbEmulatorContainer : container 9c3f1f436ec331838a8acc88bb1c32d2e041d099d42358c80e1b726dd59483bf failed to 
start
At D:\a\_tasks\CosmosDbEmulator_63c722b8-ca8a-4830-8a46-8a6b7bce9c00\2.11.28\scripts\run.ps1:100 char:13
+     $info = Start-CosmosDbEmulatorContainer -Container "$containerId" ...
+             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Start-CosmosDbEmulatorContainer

**Comment by BaherAwad** [comment]
Hello ,

on Windows Features you should "Disable" HyperV , restart you Windows
then on Windows Features "Enable" HyperV, and lastly restart you Windows.
After that Docker will work fine. 

"This works for me".
by this action you have fix/update your HyperV.
Br,

**Comment by balindra-sahani-eptura** [comment]
My problem has been short out.

**Comment by kiashok** [comment]
@fady-azmy-msft doesn't seem like anyone is actively hitting this issue anymore. Should we close this issue?

**Comment by LaurentGoderre** [comment]
I am currently get that error just running `docker run hello-world:nanoserver`

**Comment by LaurentGoderre** [comment]
I solved it by disabling Containers in "Turn Windows features on or off", restarting, re-enabling them, then restarting again

**Comment by bhavikm02** [comment]
> I solved it by disabling Containers in "Turn Windows features on or off", restarting, re-enabling them, then restarting again

I found this solution effective for running CosmosDB on a Windows Docker container.

**Comment by HackXIt** [comment]
> I solved it by disabling Containers in "Turn Windows features on or off", restarting, re-enabling them, then restarting again

I know this is closed, but for anyone else stumbling upon this thread:
The above solution did not work for me.

I was getting this error on a custom Windows IoT 21H2 image by our IT department.

Turns out this was the solution:
https://superuser.com/questions/1709644/one-of-the-hyper-v-components-is-not-running-how-to-get-more-information-on-th

**TL;DR:**
Activate _Microsoft Hyper-V Virtualization Infrastructure Driver_ in Device Manager and voilà, containers will run again.

**Comment by pBogey** [comment]
> @fady-azmy-msft doesn't seem like anyone is actively hitting this issue anymore. Should we close this issue?

I am running into this issue when issuing `docker run hello-world`.

Edition	Windows 10 Enterprise
Version	22H2
Installed on	‎15-‎Apr-‎22
OS build	19045.4894
Experience	Windows Feature Experience Pack 1000.19060.1000.0

I have both Containers and Hyper-V enabled and Microsoft Hyper-V Virtualization Infrastructure Driver is running file.

EDIT: shit, disabling and enabling HyperV did actually work. Nevermind.

**Comment by AMDphreak** [comment]
> I have confirmed Containers and HyperV are enabled.

I'd like to point out that _nothing_ in the tutorials indicates that Containers feature needs to be enabled from the Windows Features dialog. Not one word of it. I think Docker automatically enables it or something. It was not enabled on my system, until after I installed Docker.

![Image](https://github.com/user-attachments/assets/9af87b38-d467-4c92-a653-8120b6bf4177)

Also Microsoft needs to migrate their features out of the old Windows Features dialog and into the new Settings dialog.

![Image](https://github.com/user-attachments/assets/3b822b08-e706-48a7-bfc0-37f1d294f0ec)

**Comment by CaryFan** [comment]
My fix works in the following context.

I met exactly the same error when the following script during verifying that Docker Engine is installed correctly on my Windows 11 PC. 

`docker run hello-world:nanoserver`

Then I turned on the WIndows Feature <Container> and reboot my machine. Then the script works

![Image](https://github.com/user-attachments/assets/4b5b9cfe-c83c-4b55-94b6-3a4e71593c3e)
